src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/repository.py

```diff
@@ -295,58 +282,74 @@ def acr_repository_delete(registry_name,
         # Raise if --tag is empty
         if not tag:
             raise CLIError(_INVALID)
-        manifest = _delete_manifest_confirmation(yes, registry_name, repository, None, tag)
+        manifest = _delete_manifest_confirmation(
+            login_server=login_server,
+            username=username,
+            password=password,
+            repository=repository,
+            tag=tag,
+            manifest=manifest,
+            yes=yes)
         path = '/v2/{}/manifests/{}'.format(repository, manifest)
     # If --manifest is specified with a value
     else:
         # Raise if --tag is not empty
         if tag:
             raise CLIError(_INVALID)
-        manifest = _delete_manifest_confirmation(yes, registry_name, repository, manifest, None)
+        manifest = _delete_manifest_confirmation(
+            login_server=login_server,
+            username=username,
+            password=password,
+            repository=repository,
+            tag=tag,
+            manifest=manifest,
+            yes=yes)
         path = '/v2/{}/manifests/{}'.format(repository, manifest)
 
-    return _validate_user_credentials(
-        registry_name=registry_name,
-        resource_group_name=resource_group_name,
+    return _delete_data_from_registry(
+        login_server=login_server,
         path=path,
         username=username,
-        password=password,
-        repository=repository,
-        result_index=None,
-        request_method=_delete_data_from_registry
+        password=password
     )
 
 
-def _delete_manifest_confirmation(yes, registry_name, repository, manifest, tag):
-    # All tags that are referencing the manifest, this can be empty.
-    tags = []
+def _delete_manifest_confirmation(login_server,
+                                  username,
+                                  password,
+                                  repository,
+                                  tag,
+                                  manifest,
+                                  yes):
```

Comment:
`yes` is not a good parameter name. how about changes it to `confirmed`

Reply:
`yes` is the parameter exposed to users so it seems natural to use it along the path rather than creating a new name internally.

Reply:
I believe "yes" is required by the CLI for confirmation...  https://github.com/Azure/azure-cli/issues/2520

Reply:
Oh, my bad. But it is not a good name anyway. We may need to address this in the core.